### PR TITLE
Revamped author page, as per #12

### PR DESCRIPTION
**Pull request by issmirnov**
Hey,

I styled the author page, looks good both on mobile and desktop. Added a second loop partial for the post list, so that the author page doesn't have the author links on it. Added the "by Author" to the main post list, and to the post detail page.

Screenshots:
http://cloud.ivansmirnov.name/image/1R2s1R3V3e3G
http://cloud.ivansmirnov.name/image/0m0A0b0W0L3y

What do you think? The commits are all squashed and the PR is ready.

### CODE REVIEW

File: assets/scss/style.scss [hunk 1]
```diff
@@ -213,3 +215,106 @@ html, body {
...
+    /* margin-top: -1rem; */
+    margin-bottom: 1.5rem;
+    text-align: center;
+    
+    /* for separating out the author from posts */
+    padding-bottom: 2rem;
+    border-bottom: $post-border;
+}
+
+.author-location {
```
Comment (mihneadb) [inline assets/scss/style.scss]: Please remove these if they aren't used. :)

File: assets/scss/style.scss [hunk 2]
```diff
@@ -213,3 +215,106 @@ html, body {
...
+}
+
+/* author cover photo */
+.author-head {
+	background-repeat: no-repeat;
+	background-position: center center;
+	background-size: contain;
+
+	height: 30rem;
+	/* TODO - just do it based on margins*/
```
Comment (mihneadb) [inline assets/scss/style.scss]: Are these TODOs (this and the one below) still relevant?

File: assets/scss/style.scss [hunk 3]
```diff
@@ -213,3 +215,106 @@ html, body {
 section.comments {
   margin-top: 2.5rem;
 }
+
+/* === All styles related to the author page, /author/author-name. === */
+
+$img-dimension: 20rem;
+$author-padding: 6rem;
+
+.main-header {
```
Comment (mihneadb) [inline assets/scss/style.scss]: This seems unused, remove it pls.

File: assets/scss/style.scss [hunk 4]
```diff
@@ -3,12 +3,14 @@ $content-font-stack: 'Lora', 'Times New Roman', Times, serif;
 $footer-height: 15rem;
 
 $primary-color: #00807a;
-$secondary-color: #222;
-$body-color: #fefefe; // ghostwriter
+$secondary-color: #222;
```
Comment (mihneadb) [inline assets/scss/style.scss]: Please remove the trailing whitespace on this line and the one after this.

File: author.hbs [hunk 5]
```diff
@@ -1,33 +1,55 @@
...
 <section class="author-profile inner">
+
     {{#if author.image}}
-    <figure class="author-image">
-        <div class="img" style="background-image: url({{author.image}})"><span class="hidden">{{author.name}}'s Picture</span></div>
-    </figure>
+
+    <div class="row">
+	    <figure class="small-8 columns author-image">
+	        <div class="img" style="background-image: url({{author.image}})">
```
Comment (mihneadb) [inline author.hbs]: Any reason why you went with a div + bg image instead of an `img` element?

File: assets/scss/style.scss [hunk 6]
```diff
@@ -213,3 +215,106 @@ html, body {
...
+    width: $img-dimension;
+    margin: 0 auto;
+}
+
+.author-image .img {
+    margin: 0 auto;
+    height: $img-dimension;
+    background-size: cover;
+    background-position: center center;
+    border: 1px solid $primary-color;
```
Comment (mihneadb) [inline assets/scss/style.scss]: I'd drop the border, what do you think? Seems more clean.

File: author.hbs [hunk 7]
```diff
@@ -1,33 +1,55 @@
...
+    </div>
+
     {{/if}}
+
     <h1 class="author-title">{{author.name}}</h1>
-    <h2 class="author-bio">{{author.bio}}</h2>
+
+	{{#if author.bio}}
+	    <div class="row">
+		    <p class="small-10 large-10 small-centered large-centered columns author-bio">
```
Comment (mihneadb) [inline author.hbs]: Let's add a `text-left` class here, I think (especially on mobile) the bio text makes more sense when it is left aligned.

**Comment by mihneadb** [comment]
Thanks for the PR, great job! I've added some minor comments, hope you don't mind. :)

BTW, please make sure to fetch the latest changes (I merged your other PR) and re-run `gulp`, so that we can automatically merge this branch, without conflicts.


Thanks!

File: assets/scss/style.scss (hunk 3 above)
Comment (issmirnov) [inline assets/scss/style.scss]: I kept in the image dimension variable, used for the author image. Removed the padding though!

File: assets/scss/style.scss (hunk 2 above)
Comment (issmirnov) [inline assets/scss/style.scss]: Removed.

File: assets/scss/style.scss (hunk 1 above)
Comment (issmirnov) [inline assets/scss/style.scss]: Removed the empty classes.

File: assets/scss/style.scss (hunk 4 above)
Comment (issmirnov) [inline assets/scss/style.scss]: Resolved.

File: author.hbs (hunk 5 above)
Comment (issmirnov) [inline author.hbs]: Using the div allows me to use the background "cover" and "center center" properties on the div, which I don't think I can do with an img. Seems to work - it was in the file already.

File: author.hbs (hunk 7 above)
Comment (issmirnov) [inline author.hbs]: Hmm. It's a tough call - here are some screenshots of various configurations:

Desktop Centered:
center


Desktop Left:
left


Mobile Centered:
center-mobile

Mobile Left:
left-mobile


I feel that given the minimalistic theme and the centered author metadata, it looks better centered. However, the final word is yours. :)

File: assets/scss/style.scss (hunk 6 above)
Comment (issmirnov) [inline assets/scss/style.scss]: Hard to say. One one hand, definitely cleaner is better, on the other hand, if a user has a light image it's nice to have the delineation from the rest of the page. Your call.

No border: 
no-border

With border:
yes-border

**Comment by issmirnov** [comment]
I merged in your current master, rebasing my work over it. Should be able to merge now with no conflicts, once we finalize the text-align and border question.

File: author.hbs (hunk 7 above)
Comment (mihneadb) [inline author.hbs]: Fair enough, let's keep it centered then!

File: assets/scss/style.scss (hunk 6 above)
Comment (mihneadb) [inline assets/scss/style.scss]: Ok, let's keep it like that and we'll see how it looks like with other photos. We can always change it later if we want to.

**Comment by mihneadb** [comment]
Sweet, thanks so much! Sorry for the delay, been a bit busy. Merging this, great job!

**Comment by issmirnov** [comment]
Of course, glad to help! Nice job on updating the README, everything looks great. If you ever need more input/help, I'm happy to contribute if time permits. Thanks again for a beautiful theme.

**Comment by mihneadb** [comment]
You bet! :)

On Thu, Feb 19, 2015 at 11:31 PM, Ivan Smirnov <notifications@github.com>
wrote:

> Of course, glad to help! Nice job on updating the README, everything looks
> great. If you ever need more input/help, I'm happy to contribute if time
> permits. Thanks again for a beautiful theme.
>
> --
> Reply to this email directly or view it on GitHub
> <https://github.com/mihneadb/no-nonsense/pull/16#issuecomment-75140288>.
>



-- 
Mihnea Dobrescu-Balaur

**Pull request merged**